### PR TITLE
fix(agent): send the current message to the provider once

### PR DESCRIPTION
**Pull request by asim**
## Problem (downstream report, finding 1)

Every `Ask`/`Stream` sent the message being answered to the provider **twice**: memory records the user turn before the request is built (and `streamAskAI` appended it explicitly), so it arrived both as the trailing entry in `req.Messages` and as `req.Prompt` — and every provider builds its payload as `Messages` followed by `Prompt` (documented in `threadAnthropicMessages`). The model saw:

```
user:      first question
assistant: first answer
user:      the new question
user:      the new question   ← again
```

Silent, universal (default memory is store-backed), and paid for in input tokens on every turn.

## Decision

The two layers disagreed about what `Messages` means — providers read it as *history*, the agent treated it as *the full conversation* (pinned by 8 conformance/stream tests). Resolved in favor of the providers' documented contract (**option A** from the report): **`Messages` = history, `Prompt` = the current turn.**

Option C (drop `Prompt`) was considered and ruled out: the harness mock models and the groq/mistral providers require `Prompt`.

## Changes

- `requestHistory(msgs, message)` trims a **trailing** user entry equal to the current message at the two request sites (`streamAskAI`, `askLocked`'s `GenerateWithRetry` — which also covers the plan-continuation turns). Memory itself is untouched: the turn did happen and the next request needs it. A non-trailing coincidental repeat of an earlier message is never trimmed.
- The 8 tests that pinned the old contract now pin the new one — the inverse guard (`Messages` must **not** end with a duplicate of `Prompt`), in `validateConformanceRequest`, the stream conformance fake, both stream tests, and the a2a-stream-fallback harness mock.
- New regression tests (`agent/request_history_test.go`): count how many times the current message reaches the provider across both channels, with prior history, on the Ask and Stream paths (want exactly 1, history preserved), plus unit coverage of the trim edge cases.

## Verification

- `go test ./agent/ ./flow/ ./cmd/micro/` — green.
- `go run ./internal/harness/provider-conformance -providers mock` — 6/6 harnesses pass.
- gofmt clean.

## Credit

Reported by a downstream user building on go-micro v6.3.10, with a reproducer and both fix options laid out; reproduced on `master` at `225581c`. Finding 2 from the same report (the `cmd` plugin-tree link drag) lands separately.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01CmdEY7pYmV5zzwCjNJ4ykL)_

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -246,8 +274,10 @@ func (a *agentImpl) Stream(ctx context.Context, message string) (ai.Stream, erro
 		ParentID: a.parentRunID,
 		Agent:    a.opts.Name,
 	})
-	messages := append([]ai.Message(nil), a.mem.Messages()...)
-	messages = append(messages, ai.Message{Role: "user", Content: message})
+	// Messages carries the history; Prompt carries the turn being answered.
+	// Providers build their payload as Messages followed by Prompt, so
+	// appending the current message here would send it to the model twice.
+	messages := requestHistory(append([]ai.Message(nil), a.mem.Messages()...), message)
```
[the remark this answers is not in the document]
Comment (asim) [reply agent/agent.go]: Adopted in 8b832178 — you're right, and Copilot flagged the same thing: with the explicit append removed, the stream path builds its request before memory records the turn, so requestHistory there could only ever delete a legitimate older turn (e.g. a retry after an interrupted stream). Stream now passes the memory copy through untrimmed; the trim stays on the Ask path, which records before requesting. Added TestStreamPreservesIdenticalTrailingHistoryTurn to pin the edge case.

---
_Generated by [Claude Code](https://claude.ai/code)_
